clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -147,6 +158,107 @@ public void testTimeToNextUpdate_RetryBackoff() {
         assertEquals(0, metadata.timeToNextUpdate(now + 1));
     }
 
+    @Test
+    public void testIgnoreLeaderEpochInOlderMetadataResponse() {
```

Comment:
nit: Could we also briefly explain the issue in the tests? Personally, I tend to read tests to understand the expected behavior and the issue with versions earlier than 9 is not immediately apparent

Reply:
Is it OK now?